itkwidgets/integrations/environment.py

```diff
@@ -0,0 +1,24 @@
+IN_JUPYTER_NOTEBOOK = False
+IN_JUPYTERLAB = False
+IN_JUPYTERLITE = False
+IN_AWS = False
+
+try:
+    from google.colab import files
+    IN_COLAB = True
+except:
+    IN_COLAB = False
+
+if not IN_COLAB:
+    try:
+        from IPython import get_ipython
+        parent_header = get_ipython().parent_header
+        username = parent_header['header']['username']
+        if username == '':
+            IN_JUPYTERLAB = True
+        elif username == 'username':
+            IN_JUPYTER_NOTEBOOK = True
+        else:
+            IN_AWS = True
```

Comment:
This condition has not been tested as I am still unable to access Studio Lab, all other conditions have.